### PR TITLE
Device: Add Open Access Netspire OS support

### PR DESCRIPTION
**Pull request by vivia11**
Adding OS support for OA Netspire devices.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by vivia11** [comment]
@laf 

Thanks for the suggestions, I've made some changes to the code. I'm not sure if I should be unsetting my extra variables as I use them or not...

File: includes/discovery/mempools/netspire.inc.php [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+if ($device['os'] === 'netspire') {
+    echo 'Netspire Memory Pool\n';
+    $usage = str_replace('"', "", snmp_get($device, 'netSpireDeviceStorageUsed.0', "-OQv", 'OACOMMON-MIB', 'openaccess'));
```
Comment (laf) [inline includes/discovery/mempools/netspire.inc.php]: You can drop the `openaccess` from the end of the `snmp_get()` call, this is already pulled in from the yaml file.

File: includes/polling/mempools/netspire.inc.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+$usage = str_replace('"', "", snmp_get($device, "netSpireDeviceStorageUsed.0", "-OQv", 'OACOMMON-MIB', 'openaccess'));
```
Comment (laf) [inline includes/polling/mempools/netspire.inc.php]: Same here.

File: includes/polling/os/netspire.inc.php [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+$oids = array('netspireDeviceModelName.0', 'netSpireDeviceDeviceSerialNo.0');
+
+$data = snmp_get_multi_oid($device, $oids, '-OUQs', 'OACOMMON-MIB', 'openaccess');
+
+$hardware = $data[$oids[0]];
```
Comment (laf) [inline includes/polling/os/netspire.inc.php]: This is easily broken I think. Take a look at how it's done in `includes/polling/os/ipolis.inc.php`

File: includes/definitions/netspire.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+os: netspire
```
Comment (laf) [inline includes/definitions/netspire.yaml]: So netspire appears to be a range of products. Does netspire cover them all (i.e they run the same firmware)?

Comment (vivia11) [reply includes/definitions/netspire.yaml]: We have two different devices (audio controller and paging station) which run the same firmware. I am not sure if this extends to their entire range. However, their oacommon-mib covers their entire netspire range.

**Comment by laf** [comment]
Awesome work as always :)

**Pull request merged**